brainscore/benchmarks/Lee2020.py

```diff
@@ -0,0 +1,188 @@
+                bootstrap_samples_sub_assembly = int(self.bootstrap_samples * (sub_assembly.neuroid.size /
+                                                                               self._target_assembly.neuroid.size))
+
+                pairwise_tissue_corr_target.append(sample_response_corr_vs_dist(sub_assembly,
+                                                                                bootstrap_samples_sub_assembly,
+                                                                                self.ceiling))
+
+        return np.hstack(pairwise_tissue_corr_target)
+
+
+def sample_response_corr_vs_dist(assembly, num_samples, ceiling=None):
+    '''
+    1. Samples random pairs from the assembly
+    2. Computes distances for all pairs
+    3. Computes the response correlation between items of each pair
+    (4. Ceils the response correlaions)
+    :param assembly: NeuroidAssembly without stimulus repetitions
+    :param num_samples: how many random pair you want to be sampled out of the data
+    :param ceiling: if not None: expecting Score object containing ceilings of all neuroids
+    :return: [distance, pairwise_correlation_of_neuroids], pairwise correlations can be ceiled
+    '''
+    rng = np.random.default_rng()
+    neuroid_pairs = rng.integers(0, assembly.shape[0], (2, num_samples))
+
+    pairwise_distances_all = pairwise_distances(assembly)
+    pairwise_distance_samples = pairwise_distances_all[(*neuroid_pairs,)]
+
+    response_samples = assembly.data[neuroid_pairs]
+    response_correlation_samples = np.nan_to_num(corrcoef_rowwise(*response_samples))
+
+    if ceiling is not None:
+        pairwise_ceiling_all = create_pairwise_ceiling_mat(ceiling)
+        pairwise_ceiling_samples = pairwise_ceiling_all[(*neuroid_pairs,)]
+        response_correlation_samples = response_correlation_samples / pairwise_ceiling_samples
+
+    return np.vstack((pairwise_distance_samples, response_correlation_samples))
+
+
+def corrcoef_rowwise(a, b):
+    # https://stackoverflow.com/questions/41700840/correlation-of-2-time-dependent-multidimensional-signals-signal-vectors
+    a_ma = a - a.mean(1)[:, None]
+    b_mb = b - b.mean(1)[:, None]
+    ssa = np.einsum('ij,ij->i', a_ma, a_ma)  # var A
+    ssb = np.einsum('ij,ij->i', b_mb, b_mb)  # var B
+    return np.einsum('ij,ij->i', a_ma, b_mb) / np.sqrt(ssa * ssb)  # cov/sqrt(varA*varB)
+
+
+def pairwise_distances(assembly):
+    '''
+    Convenience function creating a simple lookup table for pairwise distances
+    :param assembly: NeuroidAssembly
+    :return: square matrix where each entry is the distance between the neuroids at the corresponding indices
+    '''
+    if hasattr(assembly, 'tissue_x'):
+        locations = np.stack([assembly.neuroid.tissue_x.data, assembly.neuroid.tissue_y.data]).T
+    elif hasattr(assembly, 'x'):
+        locations = np.stack([assembly.neuroid.x.data, assembly.neuroid.y.data]).T

```

Comment:
let's keep it at only one way to do things, `tissue_x`

Reply:
I am using this for both, the assembly and the model. Do you want me to add `tissue_x` to the assembly before computing? Or would you prefer to separate functions?

Reply:
ah -- yes let's set `assembly['tissue_x'] = assembly['x']` before computing and @jjpr-mit and I will try to update the assemblies accordingly